### PR TITLE
.merlin: add a SUFFIX directive for each dialect

### PR DESCRIPTION
**Pull request by vouillon**
The SUFFIX directive lets merlin know of additional file extensions.

This is helpful in our case where the source code is already valid OCaml code and can be passed to the OCaml compiler as-is. I'm not sure whether it would be possible to extend this for when a preprocessing step is needed.

### CODE REVIEW

File: src/dune_rules/merlin.ml
Comment (nojb) [inline src/dune_rules/merlin.ml:141]: We can use `Dialect.extension Dialect.ocaml Impl` instead of hardcoding `".ml"`, etc.

**Review by nojb — COMMENTED** [review]
The PR looks reasonable, thanks! I think for correctness sake we probably want to restrict to dialects that do not involve a preprocessing step, since otherwise the "jump to location" will probably land in some meaningless place.

Also it is probably a good idea to get a second pair of eyes on this, maybe @rgrinberg ?

**Comment by vouillon** [comment]
I have replaced the hard-coded default file extensions and now skip extensions with preprocessing

**Comment by nojb** [comment]
> I have replaced the hard-coded default file extensions and now skip extensions with preprocessing

Looks good now, thanks! Could you please add a `CHANGES.md` entry?

**Review by nojb — APPROVED** [review]
This is ready to be merged for me, let's leave it open for a bit to give a chance for others to chime in, otherwise am planning to merge by end of day tomorrow.

**Review by rgrinberg — APPROVED** [review]
@nojb feel free to merge

**Comment by nojb** [comment]
Apologies, while we were waiting for the CI to complete another PR got merged and now there is a tiny conflict in `CHANGES.md` that needs resolving.

**Comment by vouillon** [comment]
Thanks!

**Pull request merged**
